treesource.c

```diff
@@ -139,26 +139,49 @@ static const char *delim_end[] = {
 	[TYPE_STRING] = "",
 };
 
+/*
+ * The invariants in the marker list are:
+ *  - offsets are monotonically rising
+ *  - for a single offset there is at most one type marker
+ *  - for a single offset there is at most one non-type marker
```

Comment:
This one isn't true.
```
prop = &somenode, <&othernode>;
```
Will have the `REF_PATH` and `REF_PHANDLE` markers at the same offset, until the `REF_PATH` marker is resolved (before that it's essentially marking a region of zero length).